api/public/v2/component/views.py

```diff
@@ -38,6 +38,22 @@ def list(self, request, *args, **kwargs):
         Returns a list of components for the specified repository
         """
         commit = self.get_commit()
+        report = commit.full_report
         components = commit_components(commit, request.user)
-        serializer = ComponentSerializer(components, many=True)
+        coverage = {}
+        for component in components:
```

Comment:
if you want, you could also do this in a single pass like so:

`coverage = {}

components_with_coverage = []
for component in components:
    component_report = component_filtered_report(report, [component])
    component_coverage = round(float(component_report.totals.coverage), 2)
    coverage[component.component_id] = component_coverage

    components_with_coverage.append({
        "component_id": component.component_id,
        "name": component.name,
        "coverage": component_coverage,
    })`

Reply:
Yess! nice one